stestr/tests/test_user_config_return_codes.py

```diff
@@ -0,0 +1,444 @@
+            }, default_flow_style=False))
+        conf_file.write(contents.encode('utf-8'))
+
+        self.assertRunExit('stestr --user-config=%s run' % path, 1)
+        stream = self._get_cmd_stdout(
+            'stestr --user-config=%s last --subunit' % path)[0]
+        out, err = self.assertRunExit('stestr --user-config=%s load' % path,
+                                      0, stdin=stream)
+        out = six.text_type(out)
+        self.assertNotIn('FAILED (id=0, failures=2)', out)
+        self.assertNotIn('FF..', out)
+        self.assertIn('Totals', out)
+        self.assertIn('Worker Balance', out)
+        self.assertIn('Sum of execute time for each test:', out)
+
+    @testtools.skip('Abbreviated output not displaying')
+    def test_abbreviate_load_from_config_passing(self):
+        fd, path = tempfile.mkstemp()
+        self.addCleanup(os.remove, path)
+        conf_file = os.fdopen(fd, 'wb', 0)
+        self.addCleanup(conf_file.close)
+        contents = six.text_type(
+            yaml.dump({
+                'run': {
+                    'slowest': True,
+                },
+                'failing': {
+                    'list': True
+                },
+                'last': {
+                    'no-subunit-trace': True,
+                },
+                'load': {
+                    'abbreviate': True,
+                }
+            }, default_flow_style=False))
+        conf_file.write(contents.encode('utf-8'))
+
+        self.assertRunExit('stestr --user-config=%s run passing' % path, 0)
+        stream = self._get_cmd_stdout(
+            'stestr --user-config=%s last --subunit' % path)[0]
+        out, err = self.assertRunExit('stestr --user-config=%s load' % path,
+                                      0, stdin=stream)
+        out = six.text_type(out)
+        self.assertNotIn('PASSED (id=0)', out)
+        self.assertIn('..', out)
+        self.assertIn('Totals', out)
+        self.assertIn('Worker Balance', out)
+        self.assertIn('Sum of execute time for each test:', out)
+
+    @testtools.skip('Abbreviated output not displaying')
+    def test_abbreviate_load_from_config_failing(self):
+        fd, path = tempfile.mkstemp()
+        self.addCleanup(os.remove, path)
+        conf_file = os.fdopen(fd, 'wb', 0)
+        self.addCleanup(conf_file.close)
+        contents = six.text_type(
+            yaml.dump({
+                'run': {
+                    'slowest': True,
+                },
+                'failing': {
+                    'list': True
+                },
+                'last': {
+                    'no-subunit-trace': True,
+                },
+                'load': {
+                    'abbreviate': True,
+                }
+            }, default_flow_style=False))
+        conf_file.write(contents.encode('utf-8'))
+        # NOTE(mtreinish): Running serially here to ensure a consistent
+        # execution order for confirming the abbreviated output.
```

Comment:
I'm curious here. Can we get always the same order with the serial running? I didn't look at the doc/code yet, though.

Reply:
As far as in my local test, the order is the same every time.